aws_spot_instance_request

Terraform 0.6.0 crashes when executing the next code with an aws_spot_instance_request:

```

provider "aws" {
	region = "${var.aws_region}"
	access_key = "${var.aws_access_key}"
	secret_key = "${var.aws_secret_key}"
}


resource "aws_security_group" "default" {
	name = "terraform_sg1"
	description = "Used in terraform"
	vpc_id = "${var.vpc_id}"
	ingress {
		from_port = 0
		to_port = 65535
		protocol = "tcp"
	        security_groups = ["sg-ae3f8fcb"]
	}
        ingress {
                from_port = 8080
                to_port = 8080
                protocol = "tcp"
                cidr_blocks = ["0.0.0.0/0"]
        }
	egress {
		from_port = 0
		to_port = 0
		protocol = "-1"
		cidr_blocks = ["0.0.0.0/0"]
	}
	tags {
		Billing = "BD_DyD"
	}
}


resource "aws_spot_instance_request" "cheap_worker" {
	ami = "${var.ami_id}"
	spot_price = "0.03"
	instance_type = "t2.micro"
}
```

Exception is the next:
```
Error applying plan:

2 error(s) occurred:

* unexpected EOF
* unexpected EOF

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
panic: runtime error: invalid memory address or nil pointer dereference
2015/07/03 11:40:25 terraform-provider-aws: [signal 0xb code=0x1 addr=0x0 pc=0x50adb3]
2015/07/03 11:40:25 terraform-provider-aws: 
2015/07/03 11:40:25 terraform-provider-aws: goroutine 44 [running]:
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsSpotInstanceRequestRead(0xc2080520c0, 0xaa67c0, 0xc20802ec60, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_spot_instance_request.go:169 +0x653
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsSpotInstanceRequestUpdate(0xc2080520c0, 0xaa67c0, 0xc20802ec60, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_spot_instance_request.go:188 +0x202
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsSpotInstanceRequestCreate(0xc2080520c0, 0xaa67c0, 0xc20802ec60, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_spot_instance_request.go:130 +0xdb7
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc208152b40, 0xc2081c2ed0, 0xc208085240, 0xaa67c0, 0xc20802ec60, 0x1, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource.go:145 +0x2c1
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc208145770, 0xc2081294c0, 0xc2081c2ed0, 0xc208085240, 0xaa7960, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/provider.go:162 +0x1da
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/rpc.(*ResourceProviderServer).Apply(0xc20807cc40, 0xc208185dc0, 0xc208085550, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/rpc/resource_provider.go:323 +0x7f
2015/07/03 11:40:25 terraform-provider-aws: reflect.Value.call(0xc589c0, 0xe37c40, 0x13, 0xea2fe0, 0x4, 0xc2081c7f28, 0x3, 0x3, 0x0, 0x0, ...)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/reflect/value.go:419 +0x10e5
2015/07/03 11:40:25 terraform-provider-aws: reflect.Value.Call(0xc589c0, 0xe37c40, 0x13, 0xc2081c7f28, 0x3, 0x3, 0x0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/reflect/value.go:296 +0xbc
2015/07/03 11:40:25 terraform-provider-aws: net/rpc.(*service).call(0xc208152fc0, 0xc208152f80, 0xc20813b7f8, 0xc208076880, 0xc20819c5e0, 0xaa7900, 0xc208185dc0, 0x16, 0xaa7960, 0xc208085550, ...)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/rpc/server.go:382 +0x1f7
2015/07/03 11:40:25 terraform-provider-aws: created by net/rpc.(*Server).ServeCodec
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/rpc/server.go:476 +0x44a
2015/07/03 11:40:25 terraform-provider-aws: 
2015/07/03 11:40:25 terraform-provider-aws: goroutine 1 [IO wait]:
2015/07/03 11:40:25 terraform-provider-aws: net.(*pollDesc).Wait(0xc2080abf70, 0x72, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/fd_poll_runtime.go:84 +0x47
2015/07/03 11:40:25 terraform-provider-aws: net.(*pollDesc).WaitRead(0xc2080abf70, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/fd_poll_runtime.go:89 +0x43
2015/07/03 11:40:25 terraform-provider-aws: net.(*netFD).accept(0xc2080abf10, 0x0, 0x7fa7a655cd70, 0xc2080d6ee0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/fd_unix.go:419 +0x40b
2015/07/03 11:40:25 terraform-provider-aws: net.(*UnixListener).AcceptUnix(0xc2080dc140, 0x435000, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/unixsock_posix.go:282 +0x56
2015/07/03 11:40:25 terraform-provider-aws: net.(*UnixListener).Accept(0xc2080dc140, 0x0, 0x0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/unixsock_posix.go:293 +0x4c
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/rpc.(*Server).Accept(0xc2080d6d80, 0x7fa7a655d878, 0xc2080dc140)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/rpc/server.go:33 +0x37
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/plugin.Serve(0xc2080d6d60)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/plugin/server.go:88 +0x7e8
2015/07/03 11:40:25 terraform-provider-aws: main.main()
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/bins/provider-aws/main.go:11 +0x6d
2015/07/03 11:40:25 terraform-provider-aws: 
2015/07/03 11:40:25 terraform-provider-aws: goroutine 5 [syscall]:
2015/07/03 11:40:25 terraform-provider-aws: os/signal.loop()
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/os/signal/signal_unix.go:21 +0x1f
2015/07/03 11:40:25 terraform-provider-aws: created by os/signal.init·1
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/os/signal/signal_unix.go:27 +0x35
2015/07/03 11:40:25 terraform-provider-aws: 
2015/07/03 11:40:25 terraform-provider-aws: goroutine 7 [chan receive]:
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/plugin.func·007()
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/plugin/server.go:79 +0x69
2015/07/03 11:40:25 terraform-provider-aws: created by github.com/hashicorp/terraform/plugin.Serve
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/plugin/server.go:85 +0x7c4
2015/07/03 11:40:25 terraform-provider-aws: 
2015/07/03 11:40:25 terraform-provider-aws: goroutine 8 [select]:
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/yamux.(*Stream).Read(0xc20806d2c0, 0xc2080fc000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/yamux/stream.go:125 +0x3ab
2015/07/03 11:40:25 terraform-provider-aws: bufio.(*Reader).fill(0xc208052b40)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/bufio/bufio.go:97 +0x1ce
2015/07/03 11:40:25 terraform-provider-aws: bufio.(*Reader).Read(0xc208052b40, 0xc2080d6f80, 0x1, 0x9, 0x401192, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/bufio/bufio.go:174 +0x26c
2015/07/03 11:40:25 terraform-provider-aws: io.ReadAtLeast(0x7fa7a655e9d0, 0xc208052b40, 0xc2080d6f80, 0x1, 0x9, 0x1, 0x0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/io/io.go:298 +0xf1
2015/07/03 11:40:25 terraform-provider-aws: io.ReadFull(0x7fa7a655e9d0, 0xc208052b40, 0xc2080d6f80, 0x1, 0x9, 0x2c, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/io/io.go:316 +0x6d
2015/07/03 11:40:25 terraform-provider-aws: encoding/gob.decodeUintReader(0x7fa7a655e9d0, 0xc208052b40, 0xc2080d6f80, 0x9, 0x9, 0x0, 0x1, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/encoding/gob/decode.go:121 +0x99
2015/07/03 11:40:25 terraform-provider-aws: encoding/gob.(*Decoder).recvMessage(0xc208076780, 0xc20806e7e0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/encoding/gob/decoder.go:76 +0x55
2015/07/03 11:40:25 terraform-provider-aws: encoding/gob.(*Decoder).decodeTypeSequence(0xc208076780, 0xc2080dc800, 0x16)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/encoding/gob/decoder.go:140 +0x47
2015/07/03 11:40:25 terraform-provider-aws: encoding/gob.(*Decoder).DecodeValue(0xc208076780, 0xaae380, 0xc2080dc840, 0x16, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/encoding/gob/decoder.go:208 +0x192
2015/07/03 11:40:25 terraform-provider-aws: encoding/gob.(*Decoder).Decode(0xc208076780, 0xaae380, 0xc2080dc840, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/encoding/gob/decoder.go:185 +0x297
2015/07/03 11:40:25 terraform-provider-aws: net/rpc.(*gobServerCodec).ReadRequestHeader(0xc2080a6e10, 0xc2080dc840, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/rpc/server.go:402 +0x5e
2015/07/03 11:40:25 terraform-provider-aws: net/rpc.(*Server).readRequestHeader(0xc2080d8400, 0x7fa7a655eb30, 0xc2080a6e10, 0x0, 0x0, 0xc2080dc840, 0xc2080d6f00, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/rpc/server.go:575 +0x97
2015/07/03 11:40:25 terraform-provider-aws: net/rpc.(*Server).readRequest(0xc2080d8400, 0x7fa7a655eb30, 0xc2080a6e10, 0xc208076680, 0xc2080dc2c0, 0xacfc80, 0x0, 0x0, 0x0, 0x0, ...)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/rpc/server.go:542 +0xa2
2015/07/03 11:40:25 terraform-provider-aws: net/rpc.(*Server).ServeCodec(0xc2080d8400, 0x7fa7a655eb30, 0xc2080a6e10)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/rpc/server.go:461 +0x6d
2015/07/03 11:40:25 terraform-provider-aws: net/rpc.(*Server).ServeConn(0xc2080d8400, 0x7fa7a655ea58, 0xc20806d2c0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/rpc/server.go:453 +0x208
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/rpc.(*Server).ServeConn(0xc2080d6d80, 0x7fa7a655e970, 0xc208038118)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/rpc/server.go:76 +0x530
2015/07/03 11:40:25 terraform-provider-aws: created by github.com/hashicorp/terraform/rpc.(*Server).Accept
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/rpc/server.go:39 +0x173
2015/07/03 11:40:25 terraform-provider-aws: 
2015/07/03 11:40:25 terraform-provider-aws: goroutine 9 [IO wait]:
2015/07/03 11:40:25 terraform-provider-aws: net.(*pollDesc).Wait(0xc2080abfe0, 0x72, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/fd_poll_runtime.go:84 +0x47
2015/07/03 11:40:25 terraform-provider-aws: net.(*pollDesc).WaitRead(0xc2080abfe0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/fd_poll_runtime.go:89 +0x43
2015/07/03 11:40:25 terraform-provider-aws: net.(*netFD).Read(0xc2080abf80, 0xc2080f8000, 0x1000, 0x1000, 0x0, 0x7fa7a655cd70, 0xc20800bba8)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/fd_unix.go:242 +0x40f
2015/07/03 11:40:25 terraform-provider-aws: net.(*conn).Read(0xc208038118, 0xc2080f8000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/net.go:121 +0xdc
2015/07/03 11:40:25 terraform-provider-aws: bufio.(*Reader).fill(0xc208052720)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/bufio/bufio.go:97 +0x1ce
2015/07/03 11:40:25 terraform-provider-aws: bufio.(*Reader).Read(0xc208052720, 0xc2080d6f10, 0xc, 0xc, 0xc20800bc20, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/bufio/bufio.go:174 +0x26c
2015/07/03 11:40:25 terraform-provider-aws: io.ReadAtLeast(0x7fa7a655e9d0, 0xc208052720, 0xc2080d6f10, 0xc, 0xc, 0xc, 0x0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/io/io.go:298 +0xf1
2015/07/03 11:40:25 terraform-provider-aws: io.ReadFull(0x7fa7a655e9d0, 0xc208052720, 0xc2080d6f10, 0xc, 0xc, 0xc, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/io/io.go:316 +0x6d
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/yamux.(*Session).recvLoop(0xc2080973f0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/yamux/session.go:373 +0x10d
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/yamux.(*Session).recv(0xc2080973f0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/yamux/session.go:361 +0x28
2015/07/03 11:40:25 terraform-provider-aws: created by github.com/hashicorp/yamux.newSession
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/yamux/session.go:104 +0x499
2015/07/03 11:40:25 terraform-provider-aws: 
2015/07/03 11:40:25 terraform-provider-aws: goroutine 10 [select]:
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/yamux.(*Session).send(0xc2080973f0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/yamux/session.go:323 +0x554
2015/07/03 11:40:25 terraform-provider-aws: created by github.com/hashicorp/yamux.newSession
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/yamux/session.go:105 +0x4b3
2015/07/03 11:40:25 terraform-provider-aws: 
2015/07/03 11:40:25 terraform-provider-aws: goroutine 11 [select]:
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/yamux.(*Session).keepalive(0xc2080973f0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/yamux/session.go:279 +0x106
2015/07/03 11:40:25 terraform-provider-aws: created by github.com/hashicorp/yamux.newSession
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/yamux/session.go:107 +0x4de
2015/07/03 11:40:25 terraform-provider-aws: 
2015/07/03 11:40:25 terraform-provider-aws: goroutine 13 [select]:
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/yamux.(*Session).AcceptStream(0xc2080973f0, 0xc200000005, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/yamux/session.go:183 +0x1a8
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/rpc.(*muxBroker).Run(0xc2080dc280)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/rpc/mux_broker.go:107 +0x37
2015/07/03 11:40:25 terraform-provider-aws: created by github.com/hashicorp/terraform/rpc.(*Server).ServeConn
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/rpc/server.go:65 +0x388
2015/07/03 11:40:25 terraform-provider-aws: 
2015/07/03 11:40:25 terraform-provider-aws: goroutine 55 [select]:
2015/07/03 11:40:25 terraform-provider-aws: net/http.(*persistConn).writeLoop(0xc208097600)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/http/transport.go:945 +0x41d
2015/07/03 11:40:25 terraform-provider-aws: created by net/http.(*Transport).dialConn
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/http/transport.go:661 +0xcbc
2015/07/03 11:40:25 terraform-provider-aws: 
2015/07/03 11:40:25 terraform-provider-aws: goroutine 38 [select]:
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/yamux.(*Stream).Read(0xc20806c140, 0xc2081b8000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/yamux/stream.go:125 +0x3ab
2015/07/03 11:40:25 terraform-provider-aws: bufio.(*Reader).fill(0xc2080524e0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/bufio/bufio.go:97 +0x1ce
2015/07/03 11:40:25 terraform-provider-aws: bufio.(*Reader).Read(0xc2080524e0, 0xc20813b820, 0x1, 0x9, 0xc2080cb858, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/bufio/bufio.go:174 +0x26c
2015/07/03 11:40:25 terraform-provider-aws: io.ReadAtLeast(0x7fa7a655e9d0, 0xc2080524e0, 0xc20813b820, 0x1, 0x9, 0x1, 0x0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/io/io.go:298 +0xf1
2015/07/03 11:40:25 terraform-provider-aws: io.ReadFull(0x7fa7a655e9d0, 0xc2080524e0, 0xc20813b820, 0x1, 0x9, 0x1, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/io/io.go:316 +0x6d
2015/07/03 11:40:25 terraform-provider-aws: encoding/gob.decodeUintReader(0x7fa7a655e9d0, 0xc2080524e0, 0xc20813b820, 0x9, 0x9, 0x0, 0x1, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/encoding/gob/decode.go:121 +0x99
2015/07/03 11:40:25 terraform-provider-aws: encoding/gob.(*Decoder).recvMessage(0xc208076d00, 0xc208001e60)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/encoding/gob/decoder.go:76 +0x55
2015/07/03 11:40:25 terraform-provider-aws: encoding/gob.(*Decoder).decodeTypeSequence(0xc208076d00, 0xc20807cc00, 0x16)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/encoding/gob/decoder.go:140 +0x47
2015/07/03 11:40:25 terraform-provider-aws: encoding/gob.(*Decoder).DecodeValue(0xc208076d00, 0xaae380, 0xc20807cc60, 0x16, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/encoding/gob/decoder.go:208 +0x192
2015/07/03 11:40:25 terraform-provider-aws: encoding/gob.(*Decoder).Decode(0xc208076d00, 0xaae380, 0xc20807cc60, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/encoding/gob/decoder.go:185 +0x297
2015/07/03 11:40:25 terraform-provider-aws: net/rpc.(*gobServerCodec).ReadRequestHeader(0xc208145920, 0xc20807cc60, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/rpc/server.go:402 +0x5e
2015/07/03 11:40:25 terraform-provider-aws: net/rpc.(*Server).readRequestHeader(0xc208152f80, 0x7fa7a655eb30, 0xc208145920, 0x0, 0x0, 0xc20807cc60, 0xc20813b800, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/rpc/server.go:575 +0x97
2015/07/03 11:40:25 terraform-provider-aws: net/rpc.(*Server).readRequest(0xc208152f80, 0x7fa7a655eb30, 0xc208145920, 0xc208076880, 0xc20807d6c0, 0xaa7900, 0x0, 0x0, 0x0, 0x0, ...)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/rpc/server.go:542 +0xa2
2015/07/03 11:40:25 terraform-provider-aws: net/rpc.(*Server).ServeCodec(0xc208152f80, 0x7fa7a655eb30, 0xc208145920)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/rpc/server.go:461 +0x6d
2015/07/03 11:40:25 terraform-provider-aws: net/rpc.(*Server).ServeConn(0xc208152f80, 0x7fa7a655ea58, 0xc20806c140)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/rpc/server.go:453 +0x208
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/rpc.serve(0x7fa7a655ea58, 0xc20806c140, 0xf24330, 0x10, 0xe37ba0, 0xc20807cc40)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/rpc/server.go:146 +0x1f9
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/rpc.func·001()
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/rpc/server.go:102 +0x265
2015/07/03 11:40:25 terraform-provider-aws: created by github.com/hashicorp/terraform/rpc.(*dispenseServer).ResourceProvider
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/rpc/server.go:103 +0x107
2015/07/03 11:40:25 terraform-provider-aws: 
2015/07/03 11:40:25 terraform-provider-aws: goroutine 54 [IO wait]:
2015/07/03 11:40:25 terraform-provider-aws: net.(*pollDesc).Wait(0xc2080ab100, 0x72, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/fd_poll_runtime.go:84 +0x47
2015/07/03 11:40:25 terraform-provider-aws: net.(*pollDesc).WaitRead(0xc2080ab100, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/fd_poll_runtime.go:89 +0x43
2015/07/03 11:40:25 terraform-provider-aws: net.(*netFD).Read(0xc2080ab0a0, 0xc20811c000, 0x2000, 0x2000, 0x0, 0x7fa7a655cd70, 0xc2082245b8)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/fd_unix.go:242 +0x40f
2015/07/03 11:40:25 terraform-provider-aws: net.(*conn).Read(0xc208038fc8, 0xc20811c000, 0x2000, 0x2000, 0x0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/net.go:121 +0xdc
2015/07/03 11:40:25 terraform-provider-aws: crypto/tls.(*block).readFromUntil(0xc2081062d0, 0x7fa7a6563db8, 0xc208038fc8, 0x5, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/crypto/tls/conn.go:454 +0xe6
2015/07/03 11:40:25 terraform-provider-aws: crypto/tls.(*Conn).readRecord(0xc208060840, 0x17, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/crypto/tls/conn.go:539 +0x2da
2015/07/03 11:40:25 terraform-provider-aws: crypto/tls.(*Conn).Read(0xc208060840, 0xc208321000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/crypto/tls/conn.go:904 +0x166
2015/07/03 11:40:25 terraform-provider-aws: net/http.noteEOFReader.Read(0x7fa7a6564a18, 0xc208060840, 0xc208097658, 0xc208321000, 0x1000, 0x1000, 0xc07200, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/http/transport.go:1270 +0x6e
2015/07/03 11:40:25 terraform-provider-aws: net/http.(*noteEOFReader).Read(0xc208164ee0, 0xc208321000, 0x1000, 0x1000, 0xc208012000, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	<autogenerated>:125 +0xd4
2015/07/03 11:40:25 terraform-provider-aws: bufio.(*Reader).fill(0xc20831e4e0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/bufio/bufio.go:97 +0x1ce
2015/07/03 11:40:25 terraform-provider-aws: bufio.(*Reader).Peek(0xc20831e4e0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/bufio/bufio.go:132 +0xf0
2015/07/03 11:40:25 terraform-provider-aws: net/http.(*persistConn).readLoop(0xc208097600)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/http/transport.go:842 +0xa4
2015/07/03 11:40:25 terraform-provider-aws: created by net/http.(*Transport).dialConn
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/http/transport.go:660 +0xc9f
2015/07/03 11:40:25 terraform-provider-aws: 
2015/07/03 11:40:25 terraform-provider-aws: goroutine 49 [syscall, locked to thread]:
2015/07/03 11:40:25 terraform-provider-aws: runtime.goexit()
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/runtime/asm_amd64.s:2232 +0x1
2015/07/03 11:40:25 terraform-provider-aws: 
2015/07/03 11:40:25 terraform-provider-aws: goroutine 47 [select]:
2015/07/03 11:40:25 terraform-provider-aws: net/http.(*persistConn).roundTrip(0xc208097600, 0xc2082245d0, 0x0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/http/transport.go:1082 +0x7ad
2015/07/03 11:40:25 terraform-provider-aws: net/http.(*Transport).RoundTrip(0xc20802e000, 0xc208160dd0, 0xc208107860, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/http/transport.go:235 +0x558
2015/07/03 11:40:25 terraform-provider-aws: net/http.send(0xc208160dd0, 0x7fa7a655d050, 0xc20802e000, 0x24, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/http/client.go:219 +0x4fc
2015/07/03 11:40:25 terraform-provider-aws: net/http.(*Client).send(0x14f2d00, 0xc208160dd0, 0x24, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/http/client.go:142 +0x15b
2015/07/03 11:40:25 terraform-provider-aws: net/http.(*Client).doFollowingRedirects(0x14f2d00, 0xc208160dd0, 0x114b6a0, 0x0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/http/client.go:367 +0xb25
2015/07/03 11:40:25 terraform-provider-aws: net/http.(*Client).Do(0x14f2d00, 0xc208160dd0, 0xc208160dd0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/http/client.go:177 +0x192
2015/07/03 11:40:25 terraform-provider-aws: github.com/aws/aws-sdk-go/aws.SendHandler(0xc208310a80)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/aws/aws-sdk-go/aws/handler_functions.go:65 +0x64
2015/07/03 11:40:25 terraform-provider-aws: github.com/aws/aws-sdk-go/aws.(*HandlerList).Run(0xc208310ad0, 0xc208310a80)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/aws/aws-sdk-go/aws/handlers.go:83 +0x6e
2015/07/03 11:40:25 terraform-provider-aws: github.com/aws/aws-sdk-go/aws.(*Request).Send(0xc208310a80, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/aws/aws-sdk-go/aws/request.go:193 +0x10c
2015/07/03 11:40:25 terraform-provider-aws: github.com/aws/aws-sdk-go/service/ec2.(*EC2).DescribeSecurityGroups(0xc2080398c0, 0xc20836c120, 0x1, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/aws/aws-sdk-go/service/ec2/api.go:3644 +0x5c
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/builtin/providers/aws.func·080(0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_security_group.go:549 +0x19d
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsSecurityGroupUpdate(0xc208053440, 0xaa67c0, 0xc20802ec60, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_security_group.go:279 +0xf1
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsSecurityGroupCreate(0xc208053440, 0xaa67c0, 0xc20802ec60, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_security_group.go:246 +0x1416
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc2081529c0, 0xc2081c3ad0, 0xc20800bc50, 0xaa67c0, 0xc20802ec60, 0x1, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource.go:145 +0x2c1
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc208145770, 0xc2080d8dc0, 0xc2081c3ad0, 0xc20800bc50, 0xaa7960, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/provider.go:162 +0x1da
2015/07/03 11:40:25 terraform-provider-aws: github.com/hashicorp/terraform/rpc.(*ResourceProviderServer).Apply(0xc20807cc40, 0xc208080800, 0xc20800bfe0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/rpc/resource_provider.go:323 +0x7f
2015/07/03 11:40:25 terraform-provider-aws: reflect.Value.call(0xc589c0, 0xe37c40, 0x13, 0xea2fe0, 0x4, 0xc2081d7f28, 0x3, 0x3, 0x0, 0x0, ...)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/reflect/value.go:419 +0x10e5
2015/07/03 11:40:25 terraform-provider-aws: reflect.Value.Call(0xc589c0, 0xe37c40, 0x13, 0xc2081d7f28, 0x3, 0x3, 0x0, 0x0, 0x0)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/reflect/value.go:296 +0xbc
2015/07/03 11:40:25 terraform-provider-aws: net/rpc.(*service).call(0xc208152fc0, 0xc208152f80, 0xc20813b7f8, 0xc208076880, 0xc20807d6c0, 0xaa7900, 0xc208080800, 0x16, 0xaa7960, 0xc20800bfe0, ...)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/rpc/server.go:382 +0x1f7
2015/07/03 11:40:25 terraform-provider-aws: created by net/rpc.(*Server).ServeCodec
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/rpc/server.go:476 +0x44a
2015/07/03 11:40:25 terraform-provider-aws: 
2015/07/03 11:40:25 terraform-provider-aws: goroutine 52 [runnable]:
2015/07/03 11:40:25 terraform-provider-aws: net/http.(*persistConn).readLoop(0xc208097340)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/http/transport.go:928 +0x9ce
2015/07/03 11:40:25 terraform-provider-aws: created by net/http.(*Transport).dialConn
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/http/transport.go:660 +0xc9f
2015/07/03 11:40:25 terraform-provider-aws: 
2015/07/03 11:40:25 terraform-provider-aws: goroutine 53 [select]:
2015/07/03 11:40:25 terraform-provider-aws: net/http.(*persistConn).writeLoop(0xc208097340)
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/http/transport.go:945 +0x41d
2015/07/03 11:40:25 terraform-provider-aws: created by net/http.(*Transport).dialConn
2015/07/03 11:40:25 terraform-provider-aws: 	/opt/go/src/net/http/transport.go:661 +0xcbc
2015/07/03 11:40:25 [DEBUG] root: eval: *terraform.EvalWriteState
2015/07/03 11:40:25 [DEBUG] root: eval: *terraform.EvalApplyProvisioners
2015/07/03 11:40:25 [DEBUG] root: eval: *terraform.EvalIf
2015/07/03 11:40:25 [DEBUG] root: eval: *terraform.EvalWriteDiff
2015/07/03 11:40:25 [DEBUG] root: eval: *terraform.EvalIf
2015/07/03 11:40:25 [DEBUG] root: eval: *terraform.EvalWriteState
2015/07/03 11:40:25 [DEBUG] root: eval: *terraform.EvalApplyPost
2015/07/03 11:40:25 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* unexpected EOF
2015/07/03 11:40:25 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* unexpected EOF
2015/07/03 11:40:25 [ERROR] root: eval: *terraform.EvalOpFilter, err: 1 error(s) occurred:

* unexpected EOF
2015/07/03 11:40:25 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* unexpected EOF
2015/07/03 11:40:25 [DEBUG] root: eval: *terraform.EvalWriteState
2015/07/03 11:40:25 [DEBUG] root: eval: *terraform.EvalApplyProvisioners
2015/07/03 11:40:25 [DEBUG] root: eval: *terraform.EvalIf
2015/07/03 11:40:25 [DEBUG] root: eval: *terraform.EvalWriteDiff
2015/07/03 11:40:25 [DEBUG] root: eval: *terraform.EvalIf
2015/07/03 11:40:25 [DEBUG] root: eval: *terraform.EvalWriteState
2015/07/03 11:40:25 [DEBUG] root: eval: *terraform.EvalApplyPost
2015/07/03 11:40:25 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* unexpected EOF
2015/07/03 11:40:25 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* unexpected EOF
2015/07/03 11:40:25 [ERROR] root: eval: *terraform.EvalOpFilter, err: 1 error(s) occurred:

* unexpected EOF
2015/07/03 11:40:25 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* unexpected EOF
2015/07/03 11:40:25 [DEBUG] vertex root.root: walking
2015/07/03 11:40:25 waiting for all plugin processes to complete...
2015/07/03 11:40:25 [DEBUG] /opt/terraform/terraform-provider-aws: plugin process exited
2015/07/03 11:40:25 [DEBUG] /opt/terraform/terraform-provisioner-remote-exec: plugin process exited
2015/07/03 11:40:25 [DEBUG] /opt/terraform/terraform-provisioner-file: plugin process exited
2015/07/03 11:40:25 [DEBUG] /opt/terraform/terraform-provisioner-local-exec: plugin process exited
2015/07/03 11:40:25 [DEBUG] /opt/terraform/terraform-provisioner-chef: plugin process exited

any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.


!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

--- a/builtin/providers/aws/resource_aws_spot_instance_request.go
+++ b/builtin/providers/aws/resource_aws_spot_instance_request.go
@@ -166,7 +166,10 @@ func resourceAwsSpotInstanceRequestRead(d *schema.ResourceData, meta interface{}
 	}
 
 	d.Set("spot_bid_status", *request.Status.Code)
-	d.Set("spot_instance_id", *request.InstanceID)
+	// Instance ID is not set if the request is still pending
+	if request.InstanceID != nil {
+		d.Set("spot_instance_id", *request.InstanceID)
+	}
 	d.Set("spot_request_state", *request.State)
 	d.Set("tags", tagsToMap(request.Tags))
 

